### DISCUSSION TITLE
The *new* route folder system, how do you feel about it after using?

### DISCUSSION THREAD

**Discussion by akomm** [remix-run/remix#9483] [General]
Currently, still in dev-progress, I have 126 folders some up to 54 characters long. 

I'm almost not using file navigation for routes anymore, instead I search. It works well enough. VSCode tend to show the best match not at the top, but somewhere in the middle of the results, but so far no scrolling needed. Its because it does not weight the *order* of the words in the match to have any weight. 

I don't have a problem with the route.tsx name everywhere. I know you can now auto-label in vscode - I don't use it. Its because I find it just easy with the folder search. So far. 

Creating new route-folders in vscode is bad, because it forces you to scroll up to the route folder to create in it. I'm using a plugin with a command for that, but its still slowing down and doesn't feel good.

Technically I like the new system more. Less variants of creating same route, ability to co-locate components.

*vscode specific parts are not directly related to remix, but its important in context of usage.

**Comment by kiliman** [comment]
Other routing conventions are available. I have `remix-flat-routes,` which supports _hybrid routes_. This allows you to use folders for nesting and colocation.

For the flat-files convention, create a folder with a `+` suffix to break up long routes.

For the flat-folders convention, you can use normal folders between routes, however, the last folder is considered a colocated folder, so make sure you account for that.

https://github.com/kiliman/remix-flat-routes

**Reply by akomm** [reply]
Thanks going to check this out. I really like the *new* way for multiple reasons more than the previous one, but this one thing buggs me as I see the count of folders grow. It is by the way mostly grouped well by sorting, but some things are wild criss-cross at the top or bottom. This are spots where I would like to break down. Would probably be just 3-4 folders but a large difference.

**Reply by akomm** [reply]
In case someone stumbles here, I have tested it and it works nicely. Recommend.
